resharper/src/resharper-unity/CSharp/Daemon/Stages/Analysis/DrawGizmoAttributeProblemAnalyzer.cs

```diff
@@ -47,49 +47,43 @@ private void CheckMethodDeclaration(IMethodDeclaration methodDeclaration, IAttri
+                new[] {"component", "gizmoType"});
 
+            var match = expectedDeclaration.Match(methodDeclaration);
+            
             if (methodDeclaration.Params.ParameterDeclarations.Count == 2)
             {
                 var parameters = methodDeclaration.Params.ParameterDeclarations;
-                var expectedDeclaration = new MethodSignature(predefinedType.Void, true,
-                    new[] {componentName, gizmoTypeName},
-                    new[] {parameters[0].DeclaredName, parameters[1].DeclaredName});
 
-                if (methodDeclaration.Type.IsVoid())
-                    if (methodDeclaration.IsStatic)
-                        if (parameters[0].Type.GetTypeElement()
-                                ?.IsDescendantOf(componentName.GetTypeElement()) == true)
-                        {
-                            if (parameters[1].Type.GetTypeElement()
-                                    ?.IsDescendantOf(gizmoTypeName.GetTypeElement()) == false)
-                            {
-                                expectedDeclaration = new MethodSignature(predefinedType.Void, true,
-                                    new[] {parameters[0].Type, gizmoTypeName},
-                                    new[] {parameters[0].DeclaredName, parameters[1].DeclaredName});
-                                consumer.AddHighlighting(new IncorrectSignatureWarning(methodDeclaration,
-                                    expectedDeclaration, MethodSignatureMatch.IncorrectParameters));
-                            }
-                        }
-                        else
-                            consumer.AddHighlighting(new IncorrectSignatureWarning(methodDeclaration,
-                                expectedDeclaration, MethodSignatureMatch.IncorrectParameters));
-                    else
-                        consumer.AddHighlighting(
-                            new InvalidStaticModifierWarning(methodDeclaration, expectedDeclaration));
-                else
-                    consumer.AddHighlighting(new InvalidReturnTypeWarning(methodDeclaration, expectedDeclaration));
-            }
-            else
-            {
-                var expectedDeclaration = new MethodSignature(predefinedType.Void, true,
-                    new[] {componentName, gizmoTypeName},
-                    new[] {"scr", "gizmoType"});
-                consumer.AddHighlighting(new InvalidParametersWarning(methodDeclaration, expectedDeclaration));
+                Log.Root.Log(LoggingLevel.INFO, string.Format("component {0}:{1}:{2}", parameters[0].Type.GetTypeElement(), componentType.GetTypeElement(), parameters[0].Type.GetTypeElement()
+                    ?.IsDescendantOf(componentType.GetTypeElement())));
+                Log.Root.Log(LoggingLevel.INFO, string.Format("gizmo {0}:{1}:{2}", parameters[1].Type.GetTypeElement(), gizmoType.GetTypeElement(), parameters[1].Type.GetTypeElement()?.Equals(gizmoType?.GetTypeElement())));
+
```

Comment:
Let's lose the logs. We don't need them for release.

And for the future, you can inject `ILogger` into the constructor of components such as this and the container will provide you the correct instance, so you wouldn't use `Log.Root` directly.